arangod/Replication2/ReplicatedState/FollowerStateManager.tpp

```diff
@@ -179,44 +182,59 @@ void FollowerStateManager<S>::ingestLogData() {
+            return;
+          }
           try {
-            result.throwIfFailed();
-            LOG_TOPIC("53ba1", TRACE, Logger::REPLICATED_STATE)
-                << "leadership acknowledged - ingesting log data";
-            self->ingestLogData();
-          } catch (replicated_log::ParticipantResignedException const&) {
-            if (auto ptr = self->parent.lock(); ptr) {
-              ptr->flush(std::move(self->core));
-            } else {
-              LOG_TOPIC("15cb4", DEBUG, Logger::REPLICATED_STATE)
-                  << "LogFollower resigned, but Replicated State already "
-                     "gone";
+            try {
+              result.throwIfFailed();
+              LOG_TOPIC("53ba1", TRACE, Logger::REPLICATED_STATE)
+                  << "leadership acknowledged - ingesting log data";
+              self->ingestLogData();
+            } catch (replicated_log::ParticipantResignedException const&) {
+              if (auto ptr = self->parent.lock(); ptr) {
+                LOG_TOPIC("79e37", DEBUG, Logger::REPLICATED_STATE)
+                    << "participant resigned before leadership - force rebuild";
+                ptr->forceRebuild();
+              } else {
+                LOG_TOPIC("15cb4", DEBUG, Logger::REPLICATED_STATE)
+                    << "LogFollower resigned, but Replicated State already "
+                       "gone";
+              }
+            } catch (basics::Exception const& e) {
+              LOG_TOPIC("f2188", FATAL, Logger::REPLICATED_STATE)
+                  << "waiting for leader ack failed with unexpected exception: "
+                  << e.message();
+              FATAL_ERROR_EXIT();
             }
-          } catch (basics::Exception const& e) {
-            LOG_TOPIC("f2188", FATAL, Logger::REPLICATED_STATE)
+          } catch (std::exception const& ex) {
+            LOG_TOPIC("c7787", FATAL, Logger::REPLICATED_STATE)
                 << "waiting for leader ack failed with unexpected exception: "
-                << e.message();
+                << ex.what();
+            FATAL_ERROR_EXIT();
+          } catch (...) {
+            LOG_TOPIC("43456", FATAL, Logger::REPLICATED_STATE)
+                << "waiting for leader ack failed with unexpected exception";
             FATAL_ERROR_EXIT();
           }
-        } catch (std::exception const& ex) {
-          LOG_TOPIC("c7787", FATAL, Logger::REPLICATED_STATE)
-              << "waiting for leader ack failed with unexpected exception: "
-              << ex.what();
-          FATAL_ERROR_EXIT();
-        } catch (...) {
-          LOG_TOPIC("43456", FATAL, Logger::REPLICATED_STATE)
-              << "waiting for leader ack failed with unexpected exception";
-          FATAL_ERROR_EXIT();
-        }
-      });
+        });
+  } catch (basics::Exception const& e) {
+    if (e.code() == TRI_ERROR_REPLICATION_REPLICATED_LOG_FOLLOWER_RESIGNED) {
+      if (auto p = parent.lock(); p) {
+        return p->forceRebuild();
```

Comment:
Should we log something in this case?

Reply:
No, this is something that is expected to happen from time to time.

Reply:
Ok, I was just wondering because a few lines above we have a similar situation where we do log:
https://github.com/arangodb/arangodb/blob/2b227d7110c46c78baed5de27cf1dcf7c4b2dced/arangod/Replication2/ReplicatedState/FollowerStateManager.tpp#L93-L97